### ISSUE TITLE
스마트치안빅데이터플랫폼 기반 폐업률 예측 모델 개발

### ISSUE THREAD

**Issue by syleeie2310** [da-analysis/asac_7_dataanalysis#49]
스마트치안빅데이터플랫폼 기반 폐업률 예측 모델 개발
- y 값 : 폐업률 (정의는 은우님이 확인)
- x 변수 : 행정동, 업종, 이전 분기 점포, 연도(?)/분기()

- 모델 : regression
- 방법론 : linear regression, random forest, xgboost/lightGBM

- 채점 방식

4분기 (test 데이터 : 데이터 갯수,  23년 2분기 ~ 24년 1분기, 그 이전 기간은 train 데이터)

파리미터 튜닝 ---> train 데이터 내에서 validation set 

수요일까지 기본 모델 구성해서 test 데이터 % 나오는지? regression 기준으로 평가 스코어 계산

**Comment by jiiiim442** [comment]
# 스마트치안빅데이터플랫폼 기반 폐업률 예측 모델 개발

- [데이터 브릭스 LightGBM 링크
](https://tacademykr-asacdataanalysis.cloud.databricks.com/editor/notebooks/777952863146900?o=639069795658224)


-  [데이터 브릭스 XGBoost 링크
](https://tacademykr-asacdataanalysis.cloud.databricks.com/editor/notebooks/777952863146955?o=639069795658224#command/777952863146956)

---

# 상세 내용
- 선형회귀, RF 은우, Light GBM, XG Boost 지민 분배했습니다.
- PREV_STR_CNT (이전 분기 점포 수), CAL_CUS_RT (폐업률) 왼쪽으로 치우쳐져 있어서 로그변환 진행했습니다.
- YEAR_QTR (연도 + 분기)는 시계열로 처리했습니다.
- y train, test에 결측치가 많아 일단 제거하고 진행했습니다. (y_train 개수 1,439,688 y_train 결측치 111,398 제거)
- Light GBM RMSE: 0.9584
- XG Boost RMSE: 0.9663

**Comment by NUNURI2** [comment]
멘토님 안녕하십니까?

알려주신 내용을 바탕으로 MLflow를 활용해 모델 학습을 진행하던 중, 아래와 같은 오류가 발생하여 질문드립니다!

---

 오류 메시지: `[CONFIG_NOT_AVAILABLE] Configuration spark.mlflow.modelRegistryUri is not available.`
 SQLSTATE: `42K01`

[databricks 링크](https://tacademykr-asacdataanalysis.cloud.databricks.com/editor/notebooks/2713139915681933?o=639069795658224#command/2713139915681934)

![Image](https://github.com/user-attachments/assets/bb2c5883-760e-45ba-8fee-9e6280a7d244)

상기 문제에 대해 modelRegistryUri 관련 설정이 필요한 것인지, 아니면 Databricks 설정이 필요한 것인지 궁금합니다.

바쁘시겠지만, 여유되실 때 확인 부탁드립니다. 저희도 더 방법을 찾아보겠습니다. 감사합니다!

**Comment by NUNURI2** [comment]
# 머신러닝 진행 상황
--
[폐업률 머신러닝 결과 구글시트 링크
](https://docs.google.com/spreadsheets/d/1MVrbIk9-FpfwPSJdy9CE8aoVbobL_fKRPeQZz1fdb18/edit?gid=0#gid=0)
--

# 내용
1 사업장 개수 기준  
- 5건 이상으로 했을 때와 10건 이상으로 했을 때 분포에서 큰 차이는 관찰되지 않았습니다.
- 10건 이상으로도 충분한 데이터 수집(약50만건)이라 10건을 기준으로 잡았습니다.
- 20건 이상으로도 테스트해보려고 합니다.

2. Train/ Test 비율
- 2023년 2분기를 기준으로 했을 때 train데이터 91.07%, test 데이터 8.93%로 약 9:1로 나눴습니다.
- 2022년 1분기를 기준으로 했을 때 train데이터 80.15%, test 데이터 19.85%로 약 8:2로 나눴습니다.
- 8:2로 나눠서 실험 진행하였습니다.

3. MLflow는 아직 사용이 어려워 엑셀에 수기로 입력하였습니다.

**Comment by syleeie2310** [comment]
XGBoost로 Quantile Regression 하기 (예: lower, median, upper quantile 예측)
NGBoost 같은 분포 예측 기반 모델 사용
예측값의 분포와 실제 타깃 분포를 비교해서 왜곡된 패턴 보정

**Comment by syleeie2310** [comment]
https://xgboost.readthedocs.io/en/stable/parameter.html#learning-task-parameters
---> reg:quantileerror: Quantile loss, also known as pinball loss. See later sections for its parameter and [Quantile Regression](https://xgboost.readthedocs.io/en/stable/python/examples/quantile_regression.html#sphx-glr-python-examples-quantile-regression-py) for a worked example.

reg:tweedie: Tweedie regression with log-link. It might be useful, e.g., for modeling total loss in insurance, or for any outcome that might be [Tweedie-distributed](https://en.wikipedia.org/wiki/Tweedie_distribution#Occurrence_and_applications).

**Comment by syleeie2310** [comment]
피쳐 
- 동 단위 기준으로 데이터 갯수? (점포수 > 10개 이상 필터링) 
(업종/분기/연도...)
---> 시군구로 확장해야 될 수 있음

**Comment by syleeie2310** [comment]
모델은 xgboost로 한정해서 진행
---> 대신 파라미터 방식, 피쳐 고민

**Comment by jiiiim442** [comment]
# 머신러닝 진행 상황
[폐업률 머신러닝 결과 구글시트 링크
](https://docs.google.com/spreadsheets/d/1MVrbIk9-FpfwPSJdy9CE8aoVbobL_fKRPeQZz1fdb18/edit?gid=1635128858#gid=1635128858)

# 내용
1. 점포 개수 컷 20개로 했습니다.


2. 구 단위로 x에 넣었습니다.



3. y값을 yeo-johnson 변환 했습니다



4. Train/ Test 비율
- Train 79.01%, Test 20.99%로 나눴습니다.(2014 1분기 추가)

**Comment by syleeie2310** [comment]
@jiiiim442 gpt한테 현재 상태를 물어보고 다른 방법 (분포 결국 현재 쌍봉인데.. ) regression (Tweedie Regression)보다 나은 방식이 있을지?

없으면 이진 분류기로 최종 시도 (0 : ?% 이하 : 폐업하지 않음 : 폐업 : %이상)

---> 결과가 확률로 나올 수 있게 계산 (폐업 확률)

**Comment by jiiiim442** [comment]
# 머신러닝 진행 상황


내용
1. y의 0값을 조정 중인데 주말내로 해결해보고 안되면, 분류기로 하겠습니다.
![Image](https://github.com/user-attachments/assets/b097c59e-b8d5-4035-a993-dc358eaa96d0)

**Comment by syleeie2310** [comment]
모델 실험 완료한 이후에 

전체 데이터 ---> 확률 저장해서 어딘가에 델타 테이블로 구성해주시는 방향으로

**Comment by jiiiim442** [comment]
# 머신러닝 진행 상황
[폐업률 비0값 결과 구글시트 링크
](https://docs.google.com/spreadsheets/d/1MVrbIk9-FpfwPSJdy9CE8aoVbobL_fKRPeQZz1fdb18/edit?gid=1958158982#gid=1958158982)

---


[폐업률 분류기 결과 구글시트 링크
](https://docs.google.com/spreadsheets/d/1MVrbIk9-FpfwPSJdy9CE8aoVbobL_fKRPeQZz1fdb18/edit?gid=1150270327#gid=1150270327)

**Comment by syleeie2310** [comment]
- 분류기만 0 아닌 것들 중에서 % 정도 이하일 때 폐업 아닌 것으로 분류할지 기준 정해두고 실험 변경
- 2개 파라미터 최적화 델타 테이블 저장해서 완료하기~

**Comment by jiiiim442** [comment]
# 머신러닝 진행 상황
[폐업률 분류기 결과 구글시트 링크
](https://docs.google.com/spreadsheets/d/1MVrbIk9-FpfwPSJdy9CE8aoVbobL_fKRPeQZz1fdb18/edit?gid=5242046#gid=5242046)



--- 



[폐업률 분류기 델타 테이블 링크
](https://tacademykr-asacdataanalysis.cloud.databricks.com/explore/data/gold/ml_result_smartpolicing/%ED%8F%90%EC%97%85%EB%A5%A0_%EB%B6%84%EB%A5%98%EA%B8%B0_%EB%AA%A8%EB%8D%B8?o=639069795658224)]

**Comment by syleeie2310** [comment]
원데이터에서 모델 결과/예측값 추가해서 데이터 확인해볼 수 있는 포맷으로 델타 테이블 변경해두기

**Comment by syleeie2310** [comment]
import mlflow

mlflow.autolog(
    log_input_examples=False,
    log_model_signatures=True,
    log_models=True,
    disable=False,
    exclusive=False,
    disable_for_unsupported_versions=True,
    silent=False
)

**Comment by jiiiim442** [comment]
# 스마트치안빅데이터플랫폼 기반 폐업률 예측 모델 개발


# 상세 내용

안녕하세요 멘토님
스마트치안빅데이터플랫폼(서울시)모델 성능 높였습니다.

---

<html>
<body>
<google-sheets-html-origin><style type="text/css"></style>
모델 | Accuracy | AUC | Class 0 (폐업): Precision / Recall / F1 | Class 1 (생존): Precision / Recall / F1
-- | -- | -- | -- | --
XGBoost | 0.625 | 0.6833 | 0.59 / 0.75 / 0.66 | 0.69 / 0.50 / 0.58


</body>
</html>


---


![Image](https://github.com/user-attachments/assets/91f6ad41-8786-4594-8883-5c04f839a44f)

**Comment by syleeie2310** [comment]
네, 고생하셨어요 아직 성능이 높진 않긴한데 시간이 없으니 마무리 하는 것으로 하시죠!

**Comment by jiiiim442** [comment]
네 감사합니다!
